### ISSUE TITLE
Redirection in experiment.py vs experiment.html

### ISSUE THREAD

**Issue by awebb-nz** [nivlab/nivturk#103]
Firstly, thanks for making this library available. I am currently evaluating it to see if it would also work for our lab.

A query I have relates to how redirecting at the end of the experiment works. 

In [`experiment.py`](https://github.com/nivlab/nivturk/blob/b959fd27b5eec8e3dad0d290776b506276fb3d42/app/experiment.py#L21) you have `return redirect(url_for('complete.complete'))` which hits `/complete`, which then checks the completion state, before redirection to e.g., Prolific.

On the other hand, you also have `redirect_success` and `redirect_reject` in [`nivturk-plugins.js`](https://github.com/nivlab/nivturk/blob/b959fd27b5eec8e3dad0d290776b506276fb3d42/app/static/js/nivturk-plugins.js#L34) which redirects straight to Prolific and bypasses `/complete`.

Is there a reason for this difference, or situations where one might be preferable to the other? In my testing code, I have e.g., `redirect_success` define as:

```js
function redirect_success() {

  // Concatenate metadata into complete URL (returned on success).
  var url = "/experiment"

  $.ajax({
    url: "/redirect_success",
    method: 'POST',
    data: JSON.stringify(jsPsych.data.get().json()),
    contentType: "application/json; charset=utf-8",
  }).done(function(data, textStatus, jqXHR) {
    window.location.replace(url);
  }).fail(function(error) {
    window.location.replace(url);
  });

}
```
so it goes to `/complete` and I can handle it there.

**Comment by szorowi1** [comment]
Hi @awebb-nz,

It's very nice to meet you! Thanks so much for checking out NivTurk. 

To answer your question, I had to think back to when I first wrote that section of the code. I think my motivation was twofold:  

(1) Doing the redirect from `redirect_success()` requires fewer internal calls (e.g. `redirect_success()` --> `/experiment` --> `/complete` --> Prolific vs.  `redirect_success` --> Prolific). That said, the extra redirects are not a major cost to the app. So I don't think there's a big difference between those two options in that regard.

(2) To bypass potential latency issues. For the redirect to [`/complete` in `/experiment` to execute](https://github.com/nivlab/nivturk/blob/prolific/app/experiment.py#L18), the session needs to have been updated to reflect that the experiment is complete. That updating occurs in [`redirect_success()`](https://github.com/nivlab/nivturk/blob/prolific/app/experiment.py#L100), but not before the data is saved. However, the Ajax [`redirect_success()`](https://github.com/nivlab/nivturk/blob/prolific/app/static/js/nivturk-plugins.js#L34) call is asynchronous by default. That means it's possible, with latency, a participant is redirected back to `/experiment` before the data is finished saving and the experiment is marked as complete. At that point, they would then either have the experiment restarted or be directed to an error page (depending on if experiment restarts were allowed). 

If in your testing you find the alternative method does not run into any issues and is advantageous in some way, let me know! I'm very open to restructuring that route if there's a good reason for doing so.

Hope that helps,
Sam (@szorowi1)
